### PR TITLE
Enable cmake by default on p and zlinux

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -60,7 +60,7 @@ AC_DEFUN([OPENJ9_CONFIGURE_CMAKE],
     ],
     [
       case "$OPENJ9_PLATFORM_CODE" in
-        oa64|xa64)
```
Comment (pshipton) [inline closed/autoconf/custom-hook.m4:63]: This doesn't affect the XL builds, is that intensional?

Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4:63]: `OPENJ9_PLATFORM_CODE` is independent of normal/XL.

Comment (pshipton) [reply closed/autoconf/custom-hook.m4:63]: When I look at the nightly builds for xlinux, it seems the normal build is using cmake and the XL build is not.

Comment (dnakamura) [reply closed/autoconf/custom-hook.m4:63]: are you sure you are looking at jdk11? Looks like cmake to me https://ci.eclipse.org/openj9/job/Build_JDK11_x86-64_linux_xl_Nightly/423/console

Comment (pshipton) [reply closed/autoconf/custom-hook.m4:63]: Yes, my mistake. I was looking at jdk11 but at a plinux instead of xlinux build.

**Pull request merged**